File: leetcode/242.cpp
// Runtime: 10 ms
// Memory Usage: 7.8 MB
class Solution {
 public:
  bool isAnagram(string s, string t) {
    std::unordered_map<char, int> count;
    for (const auto& c : s) count[c] += 1;
    for (const auto& c : t) count[c] -= 1;
    for (const auto& pair : count) {
      if (pair.second != 0) return false;
    }
    return true;
  }
};

File: leetcode/242.py
"""
Runtime: 20 ms, faster than 99.99% of Python3 online submissions for Valid Anagram.
Memory Usage: 14.6 MB, less than 34.23% of Python3 online submissions for Valid Anagram.
"""
from collections import Counter


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s) == Counter(t)
